### PR TITLE
Glossary: use a table for better readability

### PR DESCRIPTION
**Pull request by mhinz**
🐻

I admit that it looks a bit weird in plaintext (and I often read Markdown in plaintext), but I think for the majority of users using a table (let alone for the alignment) improves the readability.

### CODE REVIEW

**Comment by kytrinyx** [comment]
Excellent idea 💯

File: glossary.md [hunk 1]
```diff
@@ -3,14 +3,16 @@
...
-- **language** - the name of a programming language, for example _C++_ or _Haskell_
-- **problem** - abstract concept for a programming task on exercism.io
-- **slug** - a short, unique identifier for a problem, specification, or exercise (e.g. `hamming`). Matches `/^[a-z-]+$/`
-- **solution** - a collection of iterations that a single person has done to solve an exercise
-- **specification** - a language-agnostic description of a problem. These live in [the x-common repository][x-common-repo]
-- **track** - a collection of exercises implemented in a given programming language
-- **track id** - a short, unique identifier for a track (e.g. `cpp` or `haskell`). Matches `/^[a-z-]+$/`
+| Term              | Explanation |
+|-------------------|-------------|
+| **exercise**      | a language-specific **implementation** of a **specification**. Typically consists of a test suite and a README |
```
Comment (Insti) [inline glossary.md]: implementation is bolded, but is not in the glossary.

File: glossary.md [hunk 2]
```diff
@@ -3,14 +3,16 @@
...
+|-------------------|-------------|
+| **exercise**      | a language-specific implementation of a **specification**. Typically consists of a test suite and a README |
+| **iteration**     | the code that someone writes to make a test suite pass |
+| **language**      | the name of a programming language, for example _C++_ or _Haskell_ |
+| **problem**       | abstract concept for a programming task on exercism.io |
+| **slug**          | a short, unique identifier for a **problem**, **specification**, or **exercise** (e.g. `hamming`). Matches `/^[a-z-]+$/` |
+| **solution**      | a collection of **iterations** that a single person has done to solve an **exercise** |
+| **specification** | a language-agnostic description of a **problem**. These live in [the x-common repository][x-common-repo] |
+| **track**         | a collection of **exercises** implemented in a given **language** |
+| **track id**      | a short, unique identifier for a track (e.g. `cpp` or `haskell`). Matches `/^[a-z-]+$/` |
```
Comment (kotp) [inline glossary.md]: It looks better as:

```markdown
| Term              | Explanation                                                                                                              |
|-------------------+--------------------------------------------------------------------------------------------------------------------------|
|                   | a language-specific implementation of a **specification**. Typically consists of a test suite and a README               |
| **iteration**     | the code that someone writes to make a test suite pass                                                                   |
| **language**      | the name of a programming language, for example _C++_ or _Haskell_                                                       |
| **problem**       | abstract concept for a programming task on exercism.io                                                                   |
| **slug**          | a short, unique identifier for a **problem**, **specification**, or **exercise** (e.g. `hamming`). Matches `/^[a-z-]+$/` |
| **solution**      | a collection of **iterations** that a single person has done to solve an **exercise**                                    |
| **specification** | a language-agnostic description of a **problem**. These live in [the x-common repository][x-common-repo]                 |
| **track**         | a collection of **exercises** implemented in a given **language**                                                        |
| **track id**      | a short, unique identifier for a track (e.g. `cpp` or `haskell`). Matches `/^[a-z-]+$/`                                  |
```

Comment (mhinz) [reply glossary.md]: Hmm, actually I wanted to avoid all the useless whitespace.. but I agree, it _does_ look better.

**Comment by Insti** [comment]
Waiting for this to be merged before updating the glossary based on the discussion in: https://github.com/exercism/discussions/issues/145

**Pull request merged**